### ISSUE TITLE
Payment failure messages are different in the Shortcode and Blocks checkout flows

### ISSUE THREAD

**Issue by reykjalin** [woocommerce/woocommerce-gateway-stripe#1633]
# Describe the bug

The error messages displayed when a payment fails are inconsistent across the Shortcode and Block-based checkout experiences.

# To Reproduce

1. Checkout with any of the card numbers mentioned below using both the Shortcode and Block-based checkout.
2. Note the different error messages on each page for the same card number.

# Expected behavior

- Show the same error message in both the Block-based and Shortcode-based checkout flows.

# Screenshots

| Test Number | Blocks | Shortcode |
|:---:|:---:|:---:|
| 4000000000000002 | <img width="348" alt="image" src="https://user-images.githubusercontent.com/13835680/126166093-a6480b07-6be6-4c50-b8af-cbc2a13ca2fb.png"> | <img width="500" alt="image" src="https://user-images.githubusercontent.com/13835680/126166237-585fb3d1-971a-4eb8-ba7c-7dbd4c057f04.png"> |
| 4000000000009995 | <img width="350" alt="image" src="https://user-images.githubusercontent.com/13835680/126166548-d34af903-f1d4-4e14-9615-37a111868bb4.png"> | <img width="500" alt="image" src="https://user-images.githubusercontent.com/13835680/126166393-6837df38-b612-44c3-8376-2e9cba22a9f0.png"> |
| 4000000000009987 | <img width="350" alt="image" src="https://user-images.githubusercontent.com/13835680/126166691-1278b80e-b5f2-4db4-9e33-7e5f5d96570d.png"> | <img width="500" alt="image" src="https://user-images.githubusercontent.com/13835680/126166791-6b363b40-c36e-4a8c-891e-6006b2b5cd4a.png"> |

# Suggested fix

Do one of the following:

- Use Stripe's default error messages (what we do with Blocks) without localizing them like we currently do for the Shortcode based checkout flow; or
- Map Stripe's errors to our own errors in the Block-based checkout like we currently do for the Shortcode based checkout flow.

**Comment by EvanDotPro** [comment]
We might have even more variations now with UPE! :smile:

**Comment by dannyschwantes** [comment]
I appreciate some time has passed since the original comment, thanks for your patience. 

It is possible that this issue has been fixed by a more recent release as I’ve been unable to replicate the issue you described using the latest release of the core WooCommerce plugin and the [WooCommerce Stripe Payment Gateway](https://wordpress.org/plugins/woocommerce-gateway-stripe/). 

As I’m unable to reproduce this, I’m marking this Issue as Closed. Please do open a new one if you are able to reproduce this again.
